packages/storage/src/indexed-db/updater.ts

```diff
@@ -41,10 +41,19 @@ export class IbdUpdater {
     const tables = updates.all.map(u => u.table);
     const tx = this.db.transaction(tables, 'readwrite');
 
+    const batchOperations = [];
+
+    // Batch all the updates into promises
     for (const update of updates.all) {
-      await tx.objectStore(update.table).put(update.value, update.key);
-      this.notifySubscribers(update);
+      const putOperation = tx
+        .objectStore(update.table)
+        .put(update.value, update.key)
+        .then(() => {
+          this.notifySubscribers(update);
```

Comment:
issue: we should put this below `await tx.done` for all the updates given they are not committed yet

Reply:
shouldn't we just keep the status quo here and just handle block processor atomicity separately in https://github.com/penumbra-zone/web/issues/1726?

Reply:
good call 👍